src/NServiceBus.Core/Unicast/MessageOperations.cs

```diff
@@ -11,12 +11,12 @@ namespace NServiceBus
     class MessageOperations
     {
         IMessageMapper messageMapper;
-        readonly IPipeline<IOutgoingPublishContext> publishPipeline;
-        readonly IPipeline<IOutgoingSendContext> sendPipeline;
-        readonly IPipeline<IOutgoingReplyContext> replyPipeline;
-        readonly IPipeline<ISubscribeContext> subscribePipeline;
-        readonly IPipeline<IUnsubscribeContext> unsubscribePipeline;
-        readonly IActivityFactory activityFactory;
+        protected readonly IPipeline<IOutgoingPublishContext> publishPipeline;
```

Comment:
Is there a way to avoid this? Maybe we can brainstorm something together?

Reply:
yes, I think a working alternative would be a static constructor method on the `TestableMessageOperations` class, e.g. `TestableMessageOperations.Create()` that creates the fake pipeline instances, passes them down to the ctor of the `MessagePipeline` class and assigns them to the exposed properties too. This way, no protected access would be necessary. Are you mostly concerned about the performance impact of making the fields protected?

Reply:
Not really concerned about that. I'm also not opposed to expose protected members for testability purposes. I was just wondering if there would be an elegant way to avoid them in this specific case. I spiked around for a few minutes but realized anything else is probably more complex.